tasks/build.py

```diff
@@ -1207,154 +1208,163 @@ def request_bot_build_issue_comments(repo_name, pr_number):
     # for loop because github has max 100 items per request.
     # if the pr has more than 100 comments we need to use per_page
     # argument at the moment the for loop is for a max of 400 comments could bump this up
-    for x in range(1, 5):
-        curl_cmd = f'curl -L https://api.github.com/repos/{repo_name}/issues/{pr_number}/comments?per_page=100&page={x}'
-        curl_output, curl_error, curl_exit_code = run_cmd(curl_cmd, "fetch all comments")
-
-        comments = json.loads(curl_output)
-
-        for comment in comments:
-            # iterate through the comments to find the one where the status of the build was in
-            submitted_job_comments_section = cfg[config.SECTION_SUBMITTED_JOB_COMMENTS]
-            accelerator_fmt = submitted_job_comments_section[config.SUBMITTED_JOB_COMMENTS_SETTING_WITH_ACCELERATOR]
-            instance_repo_fmt = submitted_job_comments_section[config.SUBMITTED_JOB_COMMENTS_SETTING_INSTANCE_REPO]
-            instance_repo_re = template_to_regex(instance_repo_fmt)
-            comment_body = comment['body'].split('\n')
-            instance_repo_match = re.match(instance_repo_re, comment_body[0])
-            # Check if this body starts with an initial comment from the bot (first item is always the instance + repo
-            # it is building for)
-            # Then, check that it has at least 4 lines so that we can safely index up to that number
-            if instance_repo_match and len(comment_body) >= 4:
-                # Set some defaults
-                repo_id = ""
-                on_arch = ""
-                for_arch = ""
-                date = ""
-                status = ""
-                url = ""
-                result = ""
-
-                log(f"{fn}(): found bot build response in issue, processing...")
-
-                # First, extract the repo_id
-                log(f"{fn}(): found build for repository: {instance_repo_match.group('repo_id')}")
-                repo_id = instance_repo_match.group('repo_id')
-
-                # Then, try to match the architecture we build on.
-                # First try this including accelerator, to see if one was defined
-                on_arch_fmt = submitted_job_comments_section[config.SUBMITTED_JOB_COMMENTS_SETTING_BUILD_ON_ARCH]
-                on_arch_fmt_with_accel = on_arch_fmt.format_map(PartialFormatDict(on_accelerator=accelerator_fmt))
-                on_arch_re_with_accel = template_to_regex(on_arch_fmt_with_accel)
-                on_arch_match = re.match(on_arch_re_with_accel, comment_body[1])
+
+    url = f'https://api.github.com/repos/{repo_name}/issues/{pr_number}/comments'
+    all_comments = []
+
+    try:
+        while url:
+            response = requests.get(url, params={'per_page': 100})
```

Comment:
You should specify a `timeout` argument - default is 'None', which means it'll wait forever if it doesn't get a response. I'm not sure what a good value is, but I guess 10 would be pretty reasonable.

```suggestion
            response = requests.get(url, params={'per_page': 100}, timeout=10)
```